client/src/controllers/BlockController.ts

```diff
@@ -0,0 +1,74 @@
+import { Controller } from '@hotwired/stimulus';
+
+declare global {
+  interface Window {
+    initBlockWidget?: (id: string) => void; // Declare the initBlockWidget function as optional on the Window interface
+    telepath: any;
+  }
+}
+
+/**
+ * Adds the ability to unpack a Telepath object and render it on the controlled element.
+ *
+ * @example
+ * <div
+ *  id="some-id"
+ *  data-controller="w-block"
+ *  data-w-block-data-value='{"some":"json"}'
+ * >
+ * </div>
+ */
+export class BlockController extends Controller<HTMLElement> {
+  static values = {
+    initial: { type: Array, default: [] },
+    error: { type: Object, default: {} },
+    data: { type: Object, default: {} },
+  };
```

Comment:
Been thinking about this a bit and I feel like we might get a better, simpler, solution if we change this to have TWO values only.

```
  static values = {
    args: { type: Array, default: [] },
    data: { type: Object, default: {} },
  };
```

The `initial`, and `error` could be provided as a single Stimulus value, in an array format.

Reply:
okay